src/Features/Core/Portable/GenerateMember/GenerateVariable/AbstractGenerateVariableService.cs

```diff
@@ -199,21 +199,34 @@ private static void GenerateWriteableField(ArrayBuilder<CodeAction> result, Sema
         }
 
         private static async Task AddParameterCodeActionsAsync(
-            ArrayBuilder<CodeAction> result, Document document, State state, CancellationToken cancellationToken)
+            ArrayBuilder<CodeAction> result, SemanticDocument document, State state, CancellationToken cancellationToken)
         {
             if (state.CanGenerateParameter())
             {
                 // Don't generate parameters with a `_` prefix unless that's what the user really wants as their naming style.
                 if (await NameIsHighlyUnlikelyToWarrantSymbolAsync(
-                        document, state, SymbolKind.Parameter, Accessibility.NotApplicable, cancellationToken).ConfigureAwait(false))
+                        document.Document, state, SymbolKind.Parameter, Accessibility.NotApplicable, cancellationToken).ConfigureAwait(false))
                 {
                     return;
                 }
 
-                result.Add(new GenerateParameterCodeAction(document, state, includeOverridesAndImplementations: false));
+                int? parameterIndex = null;
+                var containigMethod = state.ContainingMethod;
+
+                if (containigMethod.IsAsync && containigMethod.Parameters.Length > 0)
+                {
+                    var lastParameterIsCancellationToken = containigMethod.Parameters.LastOrDefault()!.Type.Equals(document.SemanticModel.Compilation.CancellationTokenType());
```

Comment:
```suggestion
                    var lastParameterIsCancellationToken = Equals(containigMethod.Parameters.Last().Type, document.SemanticModel.Compilation.CancellationTokenType());
```

Reply:
`Equals` is overriden for `INamedTypeSymbol`. You have told me somewhere that it is not guaranted for symbols to be always the same reference if they are equal. Won't this introduce potential bug then?

Reply:
Equals(a, b) will defer to .Equals on 'a' if 'a' is not null.  This makes things safer in case we have a parameter with a null type.  I can see the internal of IParameter.Type is:

```c#
        internal static ITypeSymbol GetPublicSymbol(this TypeWithAnnotations type)
        {
            return type.Type?.GetITypeSymbol(type.ToPublicAnnotation());
        }
```

So it def appears like it can return null.